app/models/scheduled_message.rb

```diff
@@ -6,4 +6,19 @@ class ScheduledMessage < ActiveRecord::Base
   validates :title, presence: true
 
   acts_as_taggable
+
+  def self.filter(params)
+    if params[:title].present? || params[:body].present? || params[:tag].present?
+      results = self.all.where("lower(title) like ?", "%#{params[:title].downcase}%")
+        .where("lower(body) like ?", "%#{params[:body].downcase}%")
+
+      if params[:tag].present?
+        tags = params[:tag].split(",").each { |t| t.strip }
+        results = results.tagged_with(tags, :any => true)
+      end
+      return results
```

Comment:
ah interesting this method must return `nil` when `params[:tag].present?` is false. 

well at the very least we can remove the word `return` here :)